### ISSUE TITLE
Problem with displaying the form in case of incorrect data relative to the scheme

### ISSUE THREAD

**Issue by pruchay** [json-editor/json-editor#1259]
### General information

- json-editor version: 2.9.0-beta.0

### Expected behavior
If I have a json schema, for example with one field, but given data has fields not written in json schema editor - should show that field but with label, that data is wrong.

### Actual behavior
Show null instead of information about incorrect data relative to the scheme

### Steps to reproduce the behavior
Please check replit to see configuration and example of bug
Direct link to example: https://replit.com/@pruchay/json-editorjson-editor-test#index.html

**Comment by schmunk42** [comment]
@pruchay Did you try validating your data with another tool or json-validator?

@germanbisurgi Do we have an error here internally? I guess you don't see one in the end (for the form), because the wrong data is not even applied.

CC: @robocoder

**Comment by germanbisurgi** [comment]
The data given is correct. The problem is that the type is an array `"type": ["object", "null"]` and  `"null"` is winning always against `"object"` no matter the order. When `"null"` is deleted or another type is used it works normally.  When `"additionalProperties": false` json editor ignores the new property and does not show errors.

**Comment by pruchay** [comment]
Hello @schmunk42, thank you for your answer. Yes, I checked the data with other tools. For example, I tested it with this library https://www.npmjs.com/package/jsoneditor - and with the same JSON schema and same JSON data, it works as I expected. 
Here is Repl to show it https://replit.com/@pruchay/JSON-Editor-bug-report-2#index.html - as you can see, the editor shows all data, but there is a warning with info about not correct data.

Hello @germanbisurgi, thank you for your answer. Unfortunately, I need to have a type `object` and `null`. But also my data is not null, it's the correct object, but with keys not from JSON schema. However If I delete the `null` type the editor continues to not work correctly - it begins to show me data, but only with keys from JSON schema (in this case it shows only `EURPLN`, without additional objects). I expect to show me all data with errors or data that passed validation but with errors like `should NOT have additional property`.
Also, you can check Repl that I mentioned for @schmunk42.

**Comment by schmunk42** [comment]
While our focus is on **rendering** a form rather than validation ... you'd need to at least allow `additionalProperties` for `currencies` otherwise the form can not be rendered with the additional properties.

[Example](https://json-editor.github.io/json-editor/?data=N4Ig9gDgLglmB2BnEAuUMDGCA2MBGqIAZglAIYDuApomALZUCsIANOHgFZUZQD62ZAJ5gArlELwwAJzplsrEIgwALKrNSgAJlSIx4MWAmRoAvmwAkS1epQhlUKBBQB6Zx1rwAtFbVkAdNIA5s6aUmREUJ4ADADszj6yAMQK5jCahPaOiC7OVAAeZHQQ2FR+WHTOYJzcUACMAGyMAEwAHPUttVFRfu4ICrBQJYQAKjRQAAQkUuN4IoHjUlQQ0uJsUIIQVIRVXDwKiwCOIjCL6SgA2iAYIohQ9FRSvGkgALpsZJqaBnDwcgAKUkgD1gNFQRDkiCobAggM2UhBxlA11u90ezzQIFSZxAiUBYCgzmRdwYaPSawMQ1sAGEbsSHuMAJJkkDrTaEW5SPSBBTacEibDiWwKfKFYqgi4gWpNADMABZGPUYq9oWQHA94IQAHp+ABU5gUdD0ABkqPBAlBlKhamYriIpIt4BgYOLQFjCLiwPjCXaHU7QeTBltqT7TX7kGsNkHLjsagp4Pz5G8QB8voZftgAUD4c7jFApCIoSARPojlQGVA1Ln84XIGnESAvohikJeIdjqdeDhBKg8wW2FhsAIIJCzr2qDaYVmERoQABRACqACU/kaAHIz8yLIju5zmXmIb320OCXgQMgnEAmK82vc6evXI+Ok9ni8Y1lB9i7VYgNsnKhnS5ZhPJtFg+BRIUHXgQKoMCk0nOFpwxICoJhGCzlAAZKRAAAhERgNQsCIzZWwOS5HkdDIflBRAYUCiKEpjEubooilaVlRAM81SkDVbE1TwAH5ziiTwAE4XmAWoWBaEwAApzj8F4hNE8TJNlEwAEp+P1NhDXgE0zQtK0bQg7AUNA9CWQpD8AGUqEg6DCJZSN2TzMi2F5SiBUIWjRQY1AmL8FiZXYziK24rUBKUsSJKk2T5MU4TotUjStINY1TXNS0UGta9jOUMAKF4B5ASkYwQD0MKyB4H5+msD88E9KAOTICBZX2Kgjj/TReDwE8PKosEIULRtmxPX8Oy7QbsEhNhEHywrayeHjwWmwtJF4FNvgQORT1hYEcymmbkw4Mg8kO4aYEQMg8BKIrU14XplqG9zLuu26ByHSFzpeq6bqoXapwOlAVqOxs3v+sh7RbFNvobV6/o2qGxqoaRtCkWGwYRyGwj6uyqArWHTXBxGcd4LAIG7YHnuTJGydIQFpt4O4IAx+HbuxlttBKCsNsgwEKGMEGLt+9naa5/H/oEW5WwK86ryAA) - data needs to be added manually

```
{
  "customer_id": "123",
  "currencies": {
    "EURPLN": {
      "buy_spread": "0.0150",
      "sell_spread": "0.0150"
    },
    "EURUSD": {
      "buy_spread": "0.0150",
      "sell_spread": "0.0150"
    },
    "HHHASD": {
      "buy_spread": "0.0150",
      "sell_spread": "0.0150"
    }
  }
}
```

**Comment by germanbisurgi** [comment]
As @schmunk42 already commented, JSONEditor is "edit/rendering first" and it will not set invalid values when it can. If validation errors have to shown,  a validation library like [AJV](https://ajv.js.org/guide/getting-started.html) can do the work.

In this case data validation should happen before setting the editor value.

**Comment by pruchay** [comment]
> you'd need to at least allow `additionalProperties` for currencies otherwise the form can not be rendered with the additional properties.

From my point of view, it's a wrong way. Because I don't want additional properties. 

I want to have correct JSON, that is based on defined JSON schema. Okay, this library can not validate my JSON, I can validate data using AJV, but json-editor is broke data, and looks like I can't validate data, because data is changed.

Previously we had textarea input, there user pasted JSON and one button to save data. We added JSON schema and json-editor library to make it more user-friendly. So currently we can have saved data that can be incorrect (based on JSON schema).
And now we noticed a problem: editor change data instead of showing data as is. So after user save changes, the field `currencies` will be changed to null. 

For me, it looks like bug, but from your perspective it's feature? User don't know field `currencies` real data, he can think, that it's null. If I allow `additionalProperties`, how to not allow additional properties in my JSON?

So if you consider this behavior as normal, my possibility to fix this problem is only by validating data before pasting data to json-editor and if data is wrong communicate to user without creating json-editor instance.

**Comment by schmunk42** [comment]
> From my point of view, it's a wrong way. Because I don't want additional properties.

But according to your schema these are additional properties.

Why did you explicitly disable them?

**Comment by pruchay** [comment]
> But according to your schema these are additional properties.

Yes, but 

> Why did you explicitly disable them?

This is the situation, when for some reason, stored data in DB can be incorrect. So If the form will show it to me, I will see it and fix it. Like with this library https://replit.com/@pruchay/jsoneditor-JSON-Editor-1-example#index.html - I see, that data is incorrect and I can fix it. In the case of `json-editor` I can't see that data is wrong and, which is even worse, the library for the incorrect fields it will change my object to `null` (or left only correct objects if in schema I have only `object` type).

**Comment by schmunk42** [comment]
I don't really see an easy way to make `json-editor` behave like `josdejong/jsoneditor`.

But we're open to suggestions and PRs.

**Comment by germanbisurgi** [comment]
Bug with Null schema wining fit test was fixed in #1262
